### PR TITLE
Support for sampling last `k` neighbors as temporal sampling strategy

### PR DESCRIPTION
**Pull request by rusty1s**

### CODE REVIEW

File: torch_geometric/loader/link_neighbor_loader.py [hunk 1]
```diff
@@ -101,6 +101,9 @@ class LinkNeighborLoader(LinkLoader):
             replacement. (default: :obj:`False`)
         directed (bool, optional): If set to :obj:`False`, will include all
             edges between all sampled nodes. (default: :obj:`True`)
+        temporal_strategy (string, optional): The sampling strategy when using
```
Comment (Padarn) [inline torch_geometric/loader/link_neighbor_loader.py:104]: The PR title say "last_k", is this the same thing?

Perhaps we should add some explanation of the strategies, or link to other documentation (I guess they come from pyg_lib?)

Comment (rusty1s) [reply torch_geometric/loader/link_neighbor_loader.py:104]: Good suggestion. Added more documentation.

File: torch_geometric/sampler/utils.py [hunk 2]
```diff
@@ -54,8 +54,9 @@ def to_csc(
 
     elif hasattr(data, 'adj_t'):
         if src_node_time is not None:
-            raise NotImplementedError("Temporal sampling via 'SparseTensor' "
-                                      "format not yet supported")
+            pass
+            # raise NotImplementedError("Temporal sampling via 'SparseTensor' "
```
Comment (Padarn) [inline torch_geometric/sampler/utils.py:58]: Not totally clear why we want to keep this comment

**Review by Padarn — APPROVED** [review]
This looks great, I know a few people who will be excited to see it.

**Pull request merged**